internal/handlers/pg/pgdb/query.go

```diff
@@ -247,20 +247,45 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any) {
 	var p Placeholder
 
 	for k, v := range sqlFilters.Map() {
-		if len(k) != 0 && (k[0] == '$' || types.NewPathFromString(k).Len() > 1) {
+		if len(k) != 0 && k[0] == '$' {
 			// TODO $eq and $ne https://github.com/FerretDB/FerretDB/issues/1840
-			// TODO dot notation https://github.com/FerretDB/FerretDB/issues/1841
+			// TODO $gt and $lt https://github.com/FerretDB/FerretDB/issues/1875
 			continue
 		}
 
-		// don't iterate through array for _id keys to simplify the query
-		if k == "_id" {
+		operator := "->" // operator is the operator that is used to access the field. (->/#>)
+
+		var key any = k   // key can be either a string '"v"' or path '{v,foo}'
+		var prefix string // prefix is the first key in path, if the filter key is not a path - the prefix is empty
+
+		// If the key is in dot notation use path operator (#>)
+		if len(k) != 0 {
+			if path := types.NewPathFromString(k); path.Len() > 1 {
+				operator = "#>"
+				key = path.Slice()     // '{v,foo}'
+				prefix = path.Prefix() // 'v'
+			}
+		}
+
+		// Select if value under the key is equal to provided value
+		sql := `((_jsonb%[1]s%[2]s)::jsonb = %[3]s)` // %1 - operator; %2 - key; %3 - value
+
+		// Handle _id with a simpler query, as it can't be an array
+		if k == "_id" || prefix == "_id" {
 			switch v := v.(type) {
 			case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
 				// type not supported for pushdown
+
 			case float64, string, types.ObjectID, int32, int64:
-				filters = append(filters, fmt.Sprintf(`((_jsonb->'_id')::jsonb = %s)`, p.Next()))
-				args = append(args, string(must.NotFail(pjson.MarshalSingleValue(v))))
+				filters = append(filters, fmt.Sprintf(
+					sql,      // simple query that checks for equal value under the key
+					operator, // -> or #>
+					p.Next(), // placeholder $1 used for field key or it's path for preventing SQL injections
+					p.Next(), // placeholder $2 used for field value v for preventing SQL injections
+				))
```

Comment:
```suggestion
		// Select if value under the key is equal to provided value
		sql := `((_jsonb%[1]s%[2]s)::jsonb = %[3]s)`
		
		// operator is -> for non-array, #> for array
		// placeholder p.Next() is used to prevent SQL injections.
		// The first p.Next() gets $1 and used for key,
		// the second p.Next() gets $2 and used for value.
		filters = append(filters, fmt.Sprintf(sql, operator,  p.Next(), p.Next())
```
Let's bring `sql` closer to this. It's just hard to see what operator is applied to which part of sql.

Maybe you won't like this :see_no_evil:

Reply:
Extracting comments on top of it seems much more readable, thanks!